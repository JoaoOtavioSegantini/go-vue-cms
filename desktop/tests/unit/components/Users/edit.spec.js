import { mount } from "@vue/test-utils";
import EditPage from "@/components/Users/Edit.vue";
import ClassicEditor from "@ckeditor/ckeditor5-build-classic";
import { createStore } from "vuex";

describe("EditUser.vue", () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  window.axios = {
    get: jest
      .fn(() => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosget"),
    put: jest
      .fn(() => {
        return {
          then: jest.fn(() => "your faked response"),
        };
      })
      .mockName("axiosPut"),
  };

  const mockRoute = {
    params: {
      id: 1,
    },
  };
  const mockRouter = {
    push: jest.fn(),
  };
  const createVuexStore = () => {
    return createStore({
      state() {
        return {
          Users: {
            oneUser: {
              ID: "1",
              name: "Post 1",
              email: "fake@gmail.com",
              username: "John Doe",
              password: "123",
            },
          },
        };
      },
      mutations: {
        updateUser(state) {
          state.oneUser = {
            ID: "1",
            slug: "John Doe",
            title: "Post 1",
            body: "A simple post text body",
          };
        },
      },
      actions: {
        getUser(ctx, res) {
          ctx.commit("updateUser", res);
        },
        updateUser(ctx, data) {},
      },
    });
  };

  function factory() {
    const store = createVuexStore();

    return mount(EditPage, {
      mocks: {
        axios: window.axios,
      },
      global: {
        plugins: [store],
        mocks: {
          $router: mockRouter,
          $route: mockRoute,
        },
      },
      data() {
        return {
          editor: ClassicEditor,
        };
      },
    });
  }

  it("renders props.msg when passed", async () => {
    const msg =
      "AÇÕES acbdfg@gmail.com Edição de usuário do blogNomeUsernameEmailSenhaSalvar";
    const wrapper = factory();

    wrapper.findAll("input")[0].setValue("acbdfg@gmail.com");
    wrapper.findAll("input")[1].setValue("987654");

    await wrapper.find("form").trigger("submit");
    await wrapper.find("button").trigger("click");
    await wrapper.vm.save();
    expect(mockRouter.push).toBeCalledWith({ path: "/users/1" });
    expect(wrapper.text()).toMatch(msg);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
