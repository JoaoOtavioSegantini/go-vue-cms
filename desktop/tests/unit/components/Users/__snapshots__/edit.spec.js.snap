// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditUser.vue renders props.msg when passed 1`] = `
"<div class=\\"container-fluid\\">
  <div class=\\"row\\">
    <div class=\\"col-1 nav\\">
      <p class=\\"display-6\\"><small>AÇÕES</small></p><a href=\\"/users/1\\" class=\\"btn btn-secondary\\" title=\\"Voltar\\"><i class=\\"fa fa-arrow-circle-left\\"></i></a>
    </div>
    <div class=\\"col-11\\">
      <div class=\\"card\\">
        <div class=\\"card-body\\">
          <h5 class=\\"title\\"><i class=\\"fa fa-file-text-o\\"></i> acbdfg@gmail.com <small>Edição de usuário do blog</small></h5>
          <form>
            <div class=\\"form-group\\"><label for=\\"name\\">Nome</label><input type=\\"text\\" class=\\"form-control\\" id=\\"name\\" placeholder=\\"nome do usuário\\"></div>
            <div class=\\"form-group\\"><label for=\\"name\\">Username</label><input type=\\"text\\" class=\\"form-control\\" id=\\"name\\" placeholder=\\"apelido do usuário\\"></div>
            <div class=\\"form-group\\"><label for=\\"email\\">Email</label><input type=\\"email\\" class=\\"form-control\\" id=\\"email\\" placeholder=\\"Email para autenticação\\"></div>
            <div class=\\"form-group\\"><label for=\\"password\\">Senha</label><input type=\\"password\\" class=\\"form-control\\" id=\\"password\\" placeholder=\\"Uma senha segura, informe apenas se for alterar.\\"></div><button type=\\"submit\\" class=\\"btn btn-primary\\">Salvar</button>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
